src/action_dispatcher.rs

```diff
@@ -40,6 +71,16 @@ impl ActionDispatcher {
         self.send_event(event)
     }
 
+    fn on_relative_event(&mut self, event: RelativeEvent) -> std::io::Result<()> {
+        let event = InputEvent::new_now(EventType::RELATIVE, event.code, event.value);
+        self.send_event(event)
+    }
+
+    //a function that takes mouse movement events to send in a single batch, unseparated by synchronization events.
+    pub fn send_mousemovement_event_batch(&mut self, eventbatch: Vec<InputEvent>) -> std::io::Result<()> {
```

Comment:
Can you please not make this `pub`? L27 specifically asks you: "This should be the only public method of ActionDispatcher."